### ISSUE TITLE
Kafka transport requires librdkafka

### ISSUE THREAD

**Issue by rossturk** [OpenLineage/OpenLineage#1626]
Before installing `openlineage-python[kafka]`, which contains the `conflient-kafka` client, users are required to first install the `librdkafka` C library.

This C library is not present in the Airflow Docker image, nor is it in Astro Runtime. I believe we should investigate the use of a client that does not have this underlying requirement.

**Comment by mobuchowski** [comment]
We might take a look at `pykafka`. I've used it previously and wasn't really happy with the performance, but for our use case it's a very low priority.

**Comment by mobuchowski** [comment]
@rossturk I've checked and unfortunately both non-confluent projects `pykafka` and `kafka-python` are pretty much dead. I don't think it makes sense to migrate to unmaintained projects that had last releases in 2019 or 2020.

**Comment by rossturk** [comment]
@mobuchowski I think the alternative is to document the Dockerfile steps required:

```
USER root

RUN apt-get update \
  && apt-get install -y --no-install-recommends \
    wget libssl-dev libsasl2-dev libcurl4-openssl-dev \
    software-properties-common gcc make libz-dev \
    libzstd-dev g++ \
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*

WORKDIR /root
RUN wget -qO - https://github.com/confluentinc/librdkafka/archive/refs/tags/v2.0.2.tar.gz | tar -xz
WORKDIR /root/librdkafka-2.0.2
RUN ./configure
RUN make && make check && make install
WORKDIR /opt/airflow

USER airflow
```

For distributions like Astro Runtime, we can work to convince folks to include this in their image builds.

What do you think?

**Comment by mobuchowski** [comment]
@rossturk yes, I agree - also adding `brew install librdkafka` for Mac instructions.
